frontend/src/libs/components/map/map.tsx

```diff
@@ -1,40 +1,74 @@
-import { Status, Wrapper } from '@googlemaps/react-wrapper';
+import { getValidClassNames } from '~/libs/helpers/helpers.js';
+import { useAppDispatch, useEffect, useRef } from '~/libs/hooks/hooks.js';
+import { MapService } from '~/libs/packages/map/map.js';
+import { actions as orderActions } from '~/slices/orders/order.js';
 
-import { useCallback } from '~/libs/hooks/hooks.js';
-import { config } from '~/libs/packages/config/config.js';
+import styles from './styles.module.scss';
 
-import { Spinner } from '../components.js';
-import { MapInnerComponent } from './map-inner-component/map-inner-component.js';
-
-const apiMapKey = config.ENV.API.GOOGLE_MAPS_API_KEY;
+const DEFAULT_ZOOM = 16;
 
 type Properties = {
-  center: google.maps.LatLngLiteral;
-  zoom: number;
+  center?: google.maps.LatLngLiteral;
+  zoom?: number;
   origin?: google.maps.LatLngLiteral;
   destination?: google.maps.LatLngLiteral;
   className?: string;
+  pricePerKm?: number;
+  startAddress?: string;
+  endAddress?: string;
+  onPriceChange?: (price: number) => void;
 };
 
-const Map: React.FC<Properties> = ({ ...mapProperties }) => {
-  const renderMap = useCallback(
-    (status: Status): React.ReactElement => {
-      switch (status) {
-        case Status.LOADING: {
-          return <Spinner />;
-        }
-        case Status.FAILURE: {
-          return <div>Error loading Google Maps</div>;
-        }
-        case Status.SUCCESS: {
-          return <MapInnerComponent {...mapProperties} />;
+const Map: React.FC<Properties> = ({
+  center,
+  zoom = DEFAULT_ZOOM,
+  className,
+  destination,
+  pricePerKm,
+  startAddress,
+  endAddress,
+}: Properties) => {
+  const mapReference = useRef<HTMLDivElement>(null);
+  const mapService = useRef<MapService | null>(null);
+  const mapClasses = getValidClassNames(styles.map, className);
+
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    if (mapReference.current) {
+      mapService.current = new MapService({
+        mapElement: mapReference.current,
+        center,
+        zoom,
+      });
+
+      if (center && destination) {
+        mapService.current.addMarker(destination);
+
+        void mapService.current.calculateRouteAndTime(center, destination);
+
+        if (pricePerKm && startAddress && endAddress) {
+          void dispatch(
+            orderActions.calculateOrderPrice({
+              startPoint: startAddress,
+              endPoint: endAddress,
+              pricePerKm,
+            }),
+          );
         }
       }
-    },
-    [mapProperties],
-  );
+    }
+  }, [
+    center,
+    zoom,
+    destination,
+    pricePerKm,
+    startAddress,
+    endAddress,
+    dispatch,
+  ]);
```

Comment:
I don't think that dep array should be so wide.  I think we need to trigger recalculating order price on start & end point changing. Now we will trigger price recalculation every time on zooming, centering - which is not good. Also I suppose that now pricePerKm will be consistent, so when we will add a possibility to edit it, we will add it here

And what's the difference between destination and end point?